File: Swap Nodes in Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    /**
     * @param head a ListNode
     * @return a ListNode
     */
    ListNode* swapPairs(ListNode* head) {
        // Write your code here
        ListNode dummy(INT_MIN);
        dummy.next = head;
        auto pre = &dummy;
        while (pre->next && pre->next->next)
        {
            auto first = pre->next;
            auto second = pre->next->next;
            pre->next = second;
            first->next = second->next;
            second->next = first;
            pre = pre->next->next;
        }
        return dummy.next;
    }
};

File: Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def swapPairs(self, head):
        # Write your code here
        dummy = ListNode(0)
        cur = dummy
        while head and head.next:
            tmp = head.next.next
            cur.next = head.next
            cur.next.next = head
            cur = head
            head = tmp
        cur.next = head
        return dummy.next
            